onnxruntime/contrib_ops/cpu/string_normalizer.cc

```diff
@@ -25,22 +29,85 @@ ONNX_CPU_OPERATOR_TYPED_MS_KERNEL(
 namespace string_normalizer {
 const std::string conv_error("Conversion Error");
 const std::wstring wconv_error(L"Conversion Error");
-// performs tolower/toupper in-place
-inline void ChangeCase(const std::locale& loc, StringNormalizer::CaseAction caseaction,
-                       std::wstring& wstr) {
-  assert(caseaction != StringNormalizer::NONE);
-  if (caseaction == StringNormalizer::LOWER) {
-    std::transform(wstr.begin(), wstr.end(), wstr.begin(),
-                   [&loc](wchar_t ch) { return std::tolower(ch, loc); });
-  } else {
-    std::transform(wstr.begin(), wstr.end(), wstr.begin(),
-                   [&loc](wchar_t ch) { return std::toupper(ch, loc); });
+
+// We need to specialize for MS as there is
+// a std::locale creation bug that affects different
+// environments in a different way
+#ifdef _MSC_VER
+
+class Locale {
+ public:
+  explicit Locale(const std::string& name)
+      : loc_(nullptr) {
+    loc_ = _create_locale(LC_CTYPE, name.c_str());
+    if (loc_ == nullptr) {
+      ONNXRUNTIME_THROW("Failed to construct locale with name:",
+                        name, ":", ":Please, install necessary language-pack-XX and configure locales");
+    }
+  }
+
+  ~Locale() {
+    if (loc_ != nullptr) {
+      _free_locale(loc_);
+    }
   }
-}
+
+  Locale(const Locale&) = delete;
```

Comment:
you can use ONNXRUNTIME_DISALLOW_COPY_ASSIGNMENT_AND_MOVE macro.

Reply:
Would it be possible to fix the macro and make those `= delete` public?

Reply:
you can place the macro in the public section of your class.